sycl/doc/extensions/experimental/sycl_ext_oneapi_matrix/sycl_ext_oneapi_matrix.asciidoc

```diff
@@ -110,157 +145,204 @@ enum class layout {
+is supported.
 
-#### Load
+==== Load
 ```c++
 namespace sycl::ext::oneapi::experimental::matrix {
-  template <typename Group, typename T, size_t NumRows, size_t NumCols,
+  template <typename Group, typename T, typename S,
+            size_t NumRows, size_t NumCols,
             access::address_space Space>
   void joint_matrix_load(Group sg,
-    joint_matrix<Group, T, use::accumulator, NumRows, NumCols, layout::dynamic> &res,
-    multi_ptr<T, Space, IsDecorated> src, size_t stride, layout Layout);
+    joint_matrix<Group, T, use::accumulator, NumRows, NumCols,
+    layout::dynamic> &res,
+    multi_ptr<S, Space, IsDecorated> src, size_t stride, layout Layout);
     
-  template <typename Group, typename T, size_t NumRows, size_t NumCols,
-          use Use, layout Layout, access::address_space Space>
+  template <typename Group, typename T, typename S,
+            size_t NumRows, size_t NumCols,
+            use Use, layout Layout, access::address_space Space>
   void joint_matrix_load(Group sg,
     joint_matrix<Group, T, Use, NumRows, NumCols, Layout> &res,
-    multi_ptr<T, Space, IsDecorated> src, size_t stride);
+    multi_ptr<S, Space, IsDecorated> src, size_t stride);
 }
 ```
 
-`joint_matrix_load` loads data from memory to the 2d tiles/registers of the tensor hardware.
-We define two overloads of the load function depending on whether the memory layout was declared as part of the `joint_matrix` type or not. 
-The first overload that takes memory layout as an argument is only available for a `joint_matrix` type that used the default value `layout::dynamic`.
-The second overload without a memory layout must not be used with a `joint_matrix` type that used the default value `layout::dynamic`.
-
-The base pointer `src` here determines the starting address of the matrix to be loaded from. `Layout` determines whether the data is being read in a row (`row_major`), column major (`column_major`) fashion. `stride` describes the number of elements between consecutive rows for the row major layout, or between columns for the column major layout. 
-
-
-#### Store
+`joint_matrix_load` loads data from memory to the 2d tiles/registers
+of the matrix hardware.
+We define two overloads of the load function depending on whether the
+memory layout was declared as part of the `joint_matrix` type or not. 
+The first overload that takes memory layout as an argument is only
+available for a `joint_matrix` type that used the default value
+`layout::dynamic`.
+The second overload without a memory layout must not be used with a
+`joint_matrix` type that used the default value `layout::dynamic`.
+
+The base pointer `src` here determines the starting address of the
+matrix to be loaded from. `Layout` determines whether the data is
+being read in a row (`row_major`), column major (`column_major`)
+fashion. `stride` describes the number of elements between consecutive
+rows for the row major layout, or between columns for the column major
+layout.
```

Comment:
Right, this is specific to the tf32 case where the elements of memory are float but type of joint matrix is precision::tf32. SO these can be different.
In the store case, since we only supports store of accumulator matrix which is always full precision, so no separation of types there.

Reply:
> In the store case, since we only supports store of accumulator matrix which is always full precision

I didn't realize this.  Does the spec say someplace that `tf32` is not allowed for the accumulator matrix?

> Right, this is specific to the tf32 case where the elements of memory are float but type of joint matrix is precision::tf32.

If this is the only reason, then I don't think you need separate `T` and `S` template parameters.  You could instead use the "alias" mechanism you describe in your other comment, or you can define separate overloads for the case when `T` is `tf32`.

Reply:
@JackAKirk, can you please confirm this is the only reason for the CUDA backend to have type of matrix different from type of memory elements?
if this is the case, the recommendation is to create overloads for tf32. @JackAKirk, Is that okay for the CUDA backend as well?